sklearn/feature_selection/_from_model.py

```diff
@@ -270,12 +347,25 @@ def _get_support_mask(self):
                 "instead."
             )
 
-        scores = _get_feature_importances(
-            estimator=estimator,
-            getter=self.importance_getter,
-            transform_func="norm",
-            norm_order=self.norm_order,
-        )
+        if self.importance_type == "model":
+            scores = _get_feature_importances(
+                estimator=estimator,
+                getter=self.importance_getter,
+                transform_func="norm",
+                norm_order=self.norm_order,
+            )
+        elif self.importance_type == "permutation":
+            if not hasattr(self, "_score"):
+                raise NotFittedError(
+                    "The `fit` method must be called to calculate feature importance"
+                    " when using permutation importance."
+                )
+            scores = self._score
+        else:
+            raise ValueError(
+                "Only `model` and `permutation` is supported for importance_type."
```

Comment:
Hi @MaxwellLZH, thanks for your pull request and for your patience.
The error raised here is not tested. Do you mind adding the related test and synchronizing with upstream?
This will help next reviewers. Thanks!